src/Stan/FileInfo.hs

```diff
@@ -41,3 +46,12 @@ extensionsToText = \case
         case parsedExtensionsSafe of
             Just s  -> show s : exts
             Nothing -> exts
+
+{- | Check whether the given extension is disabled
+-}
+isExtensionDisabled :: Extension -> ExtensionsResult -> Bool
+isExtensionDisabled ext = \case
+    Left _ -> True  -- no info about extensions, consider it disabled
+    Right Extensions{..} ->
+           Set.member (Off ext) extensionsAll
+        || Set.notMember (On ext) extensionsAll
```

Comment:
I guess the second case is more likely to be there, so I think it makes sense to check it first